File: Capes.WEB/src/app/domain/filter-result.ts
import { IBook } from './book';
import { IAward } from './award';
import { IParticipation } from './participation';
import { IPublication } from './publication';

export interface IFilterResult {
    books?: IBook[];
    awards?: IAward[];
    participations?: IParticipation[];
    publications?: IPublication[];
};

File: Capes.API/Models/FilterResult.cs
using Capes.Domain;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Capes.API.Models
{
    public class FilterResult
    {
        [JsonProperty("books")]
        public List<Book> Books { get; set; }

        [JsonProperty("awards")]
        public List<Award> Awards { get; set; }

        [JsonProperty("participations")]
        public List<Participation> Participations { get; set; }

        [JsonProperty("publications")]
        public List<Publication> Publications { get; set; }
    }
}